Core/Core/Files/APIFile.swift

```diff
@@ -193,6 +194,18 @@ public struct APIUsageRights: Codable, Equatable {
     public let use_justification: UseJustification?
 }
 
+public enum APIFolderItem: Codable {
+    case file(APIFile)
+    case folder(APIFolder)
+
+    public init(from decoder: Decoder) throws {
+        throw NSError.instructureError("Not for actual api decoding")
+    }
+    public func encode(to encoder: Encoder) throws {
+        throw NSError.instructureError("Not for actual api encoding")
```

Comment:
We shouldn't use `APIFolderItem` as a `Response` type, but it need to be `Codable` for the use case to use it.